emu-scripts/F-ZeroScreen.lua

```diff
@@ -0,0 +1,445 @@
+
+--Function: Get_Inputs_For_AI
+--Description: Setup the AI input values
+--Inputs:
+--	Depends on Vehicle_Direction_Angles being set
+--	Depends on Screen variables to be set
+--Outputs:
+--  AI_Inputs - Inputs for the AI 
+--				The Input are defined as an array of tile values {1, 2 .. 224, 225} where 1 is (0,0) of the top left of the
+--				and 2 is (0,1) of the screen and 224 is (14, 15) and 225 is (15, 15), if 15 is the screen's width and heigth
+function Get_Inputs_For_AI()
+	--console.log("Starting Input Creation")
+	
+	local AI_Inputs = {}
+	
+	-- Calculate vectors for the vehicles direction
+	local Current_Direction = {math.sin(math.rad(Vehicle_Direction_Angles)), -math.cos(math.rad(Vehicle_Direction_Angles))}
+	
+	--console.log("Get_Inputs_For_AI local variable Current_Direction =")
+	--console.log(Current_Direction)
+	
+	-- Reset AI_Inputs_Index for screen displaying
+	local AI_Inputs_Index = 1
+	local Tile_Color = 1
+	
+	-- Loop through the screens Heigth and width to get the tile value for the AI_Input variables.
+	for Current_Tile_Row = Global_Screen_Heigth, 1, -1 do
+		for Current_Tile_Column = -Half_Width, Half_Width do
+			AI_Inputs[#AI_Inputs+1] = Get_Tile(Current_Tile_Row, Current_Tile_Column, Current_Direction)
+			
+			--Draw As we go
+			--Compilier requirements
+			if Tile_Color ~= nil then
+				--console.log("Trying to get tile color")	
+			    --console.log("Tile is =")
+				--console.log(AI_Inputs[AI_Inputs_Index])
+				
+				-- Set a local variable for the current tiles value
+				Tile_Color = Get_Tile_Value(AI_Inputs[AI_Inputs_Index])
+				
+				--console.log("Tile_Color is =")
+				--console.log(Tile_Color)
+				
+				-- Set up Tile_Color to be grayscaled.
+				--Compilier requirements
+				if Tile_Color ~= nil then 
+					Tile_Color = math.floor((Tile_Color+2)/4*255)
+				end
+			end
+			--Compilier requirements
+			if Tile_Color ~= nil then
+				--Grayscale value to display.
+				Tile_Color = Tile_Color + Tile_Color*0x100 + Tile_Color*0x10000 + 0xFF*0x1000000
+				--console.log("Got tile color")
+			end
+			
+			--Since all maps are rotated, we rotate back to make the screen display make since.
+			Rotated_Coordinates = rotate(Current_Tile_Row, Current_Tile_Column, 7.5, 7.5, 270)
+			
+			--Draw on the screen the current Tile values.
+			gui.drawPixel(Rotated_Coordinates[1]+10, Rotated_Coordinates[2]+10, Tile_Color)
+			AI_Inputs_Index = AI_Inputs_Index + 1
+		end
+	end
+	
+	--console.log("Stoping Input Creation")
+	--console.log("AI_Inputs =")
+	--console.log(AI_Inputs)
+	
+	--AI_Input array return
+	return AI_Inputs
+end
+
+--Function: Start_Mapping
+--Description: Function to call from outside of F-zeroScreen to start input calculations.
+--Inputs:
+--  NA
+--Outputs:
+--  NA
+function F-zeroScreen.Start_Mapping()
+--Get Car positions
+    Get_Vehicle_Details()
+
+--Calculate angle of car
+    Get_Car_Dircection_In_Degrees()
+	
+--GetInputs
+    local Inputs_For_AI = {}
+	Inputs_For_AI = Get_Inputs_For_AI()
+
```

Comment:
```suggestion
return Inputs_For_AI
```
This needs to return the inputs so we can use them in the main execution loop.
Will everything here work if we just call this function for every frame in the main execution loop?

Reply:
Yes. I have it working in a play.lua to send to the AI